core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java

```diff
@@ -117,77 +126,106 @@ private CarbonCommonConstants() {
    * CARBON_TIMESTAMP
    */
   public static final String CARBON_TIMESTAMP_MILLIS = "dd-MM-yyyy HH:mm:ss:SSS";
+
   /**
-   * compressor for writing/reading carbondata file
+   * Property for specifying the format of DATE data type column.
+   * e.g. yyyy/MM/dd , or using default value
    */
+  @CarbonProperty
+  public static final String CARBON_DATE_FORMAT = "carbon.date.format";
+
+  /**
+   * default value
+   */
+  public static final String CARBON_DATE_DEFAULT_FORMAT = "yyyy-MM-dd";
+
+  /**
+   * compressor for writing/reading CarbonData file
+   */
+  @CarbonProperty
   public static final String COMPRESSOR = "carbon.column.compressor";
 
   /**
    * default compressor is snappy
    */
   public static final String DEFAULT_COMPRESSOR = "snappy";
+
   /**
-   * ZOOKEEPER_ENABLE_LOCK if this is set to true then zookeeper will be used to handle locking
+   * ZOOKEEPER_ENABLE_LOCK if this is set to true then zookeeper
+   * will be used to handle locking
    * mechanism of carbon
    */
   @CarbonProperty
   public static final String LOCK_TYPE = "carbon.lock.type";
 
+  /**
+   * ZOOKEEPER_ENABLE_DEFAULT the default value for zookeeper will be true for carbon
+   */
+  public static final String LOCK_TYPE_DEFAULT = "LOCALLOCK";
+
   /**
    * Specifies the path where the lock files have to be created.
    * By default, lock files are created in table path.
    */
   @CarbonProperty
   public static final String LOCK_PATH = "carbon.lock.path";
 
-  /**
-   * ZOOKEEPER_ENABLE_DEFAULT the default value for zookeeper will be true for carbon
-   */
-  public static final String LOCK_TYPE_DEFAULT = "LOCALLOCK";
+  public static final String LOCK_PATH_DEFAULT = "";
 
   /**
    * ZOOKEEPER_LOCATION this is the location in zookeeper file system where locks are created.
    * mechanism of carbon
    */
   public static final String ZOOKEEPER_LOCATION = "/CarbonLocks";
+
   /**
-   * xxhash algorithm property for hashmap
+   * xxhash algorithm property for hashMap
```

Comment:
keep `hashmap`, and can you add comment to describe in where this is used for? query/loading/compaction?

Reply:
ok,done. We can add it in next PR, add .usedBy("") for it.